Homework2/Trie/Trie.cs

```diff
@@ -0,0 +1,260 @@
+﻿namespace Trie;
+using System.Collections;
+
+/// <summary>
+/// Trie-structure.
+/// </summary>
+public class Trie
+{
+    private TrieElement head = new();
+    private TrieElement? triePointer = new();
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="Trie"/> class.
+    /// </summary>
+    public Trie()
+    {
+        this.triePointer = this.head;
+    }
+
+    /// <summary>
+    /// Gets the size of the TrieElement.
+    /// </summary>
+    public int Size
+    {
+        get { return this.head.SizeOfTrieElement; }
+    }
+
+    /// <summary>
+    /// Method adds the element into Trie.
+    /// </summary>
+    /// <param name="stringElement">The string type element to adding.</param>
+    /// <returns>True if the stringElement hasn't been previously consisted in Trie. Else false.</returns>
+    public bool Add(string stringElement)
+    {
+        if (this.Contains(stringElement) || stringElement.Length == 0 || stringElement == null)
+        {
+            return false;
+        }
+
+        var stringElementIndex = 0;
+        TrieElement? currentElement = this.head;
+        ++currentElement.SizeOfTrieElement;
+        while (currentElement != null && currentElement.Vertexes.ContainsKey(stringElement[stringElementIndex]))
+        {
+            currentElement = currentElement.Vertexes[stringElement[stringElementIndex]] as TrieElement;
+            ++currentElement!.SizeOfTrieElement;
+            if (stringElementIndex == stringElement.Length - 1)
+            {
+                if (!currentElement!.IsTerminal)
+                {
+                    return currentElement.IsTerminal = true;
```

Comment:
Выражение с побочным эффектом, нехорошо. Лучше отдельно:
```suggestion
                    currentElement.IsTerminal = true;
                    return true;
```